OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -350,7 +350,7 @@ public override void Cancel(Actor self, bool keepQueue = false)
 		{
 			// We need to clear the path here in order to prevent MovePart queueing new instances of itself
 			// when the unit is making a turn.
-			if (path != null && mobile.CanStayInCell(mobile.ToCell))
```

Comment:
I really don't like the idea of this change - I would be willing to bet that this will create more cases where units end up sitting on transit-only tiles or in tunnels, rather than less.

`Move` is often queued as a child activity, and may be cancelled by the parent activity without the unit going idle, meaning that the change in `OnBecomingIdle` won't be able to move the unit out of the cell.

Reply:
How do you suggest the locking situation be resolved then? The problem occurs when the destination is not reachable due to being blocked, but the current tile is transit-only and therefore refuses to cancel.

I wanted to do this by adding a check-forwards when crossing transit-only to block/re-route if the exit is not currently clear, and being able to catch the case and allow the order to cancel even if it is on a transit-only space when there is no appropriate exit appears to be necessary too.

The main issue I'd see with the change as proposed is that it wouldn't stop a unit deploying on a transit-only space unless they explicitly disallowed it.

Reply:
We won't be able to fix this properly without implementing more of the docking logic. The simplest workaround would still be to revert the service depots back to a blocking footprint type.

Otherwise, the simplest workaround that may have to be good enough will be to add some additional logic around the `// HACK` and `// TODO` comments in `Resupply` to make mobile units for do a `MoveAdjacentTo` and then only `MoveWithinRange` / `MoveTo` if they are on an adjacent cell and the target cell is not occupied.

Reply:
be4a1a59c1dffb539646fbb60d30dc8b02695bde illustrates a way we could implement this for Resupply without regressing other cases. It is a fairly awful hack, but no worse than the rest of the Resupply code.

Reply:
I'll have a look into doing the "proper" fix tomorrow.